libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -617,16 +622,22 @@ data Specification fn a where
   -- | Anything
   TrueSpec :: Specification fn a
 
+instance Arbitrary a => Arbitrary (NE.NonEmpty a) where
+  arbitrary = do
+    NonEmpty xs <- arbitrary
```

Comment:
Is this the `NonEmpty` from QuickCheck? 
Alternatively , you could use the one that is already implemented in `import Test.QuickCheck.Instances ()`, if you add  `quickcheck-instances` as dependency.

Reply:
I am not sure. But it is the Standard way to use non empty lists.